nca/NetworkConfig/NetworkLayer.py

```diff
@@ -229,6 +229,9 @@ def captured_cond_func(policy):
         if not captured_res:  # no allow policies for target
             # add connections allowed by default that are not captured
             allowed_non_captured_conns |= (ConnectionSet(True) - denied_conns)
+            # exception: update allowed non-captured conns to DNSEntry dst with TCP only
+            if isinstance(to_peer, DNSEntry):
+                allowed_non_captured_conns = ConnectionSet.get_all_tcp_connections()
```

Comment:
why is it only in non-captured conns? isn't it also relevant for connections returned from `IstioSidecar.allowed_connections()`?

Reply:
it is , didn't see a need to currently change since sidecar is an istio policy (then considered automatically as TCP only )

Reply:
where is it considered as TCP only? It seems that `IstioSidecar.allowed_connections()` returns `All connections`, and that `IstioNetworkLayer._allowed_xgress_conns()` returns `All connections` for connections captured to `DNSEntry` and `TCP` for connections not captured to `DNSEntry`?

Reply:
oh, I assumed by splitting to tcp and non-tcp in the ConnectivityMapQuery, all istio's layer conns are TCP 
committed a fix

Reply:
still not sure I understand, how did it work (passed the tests) before this fix? 
and why do you restrict sidecar connections to `TCP` only for `DNSentry`?

Reply:
I agree, I assumed that it worked previously because by splitting connections to TCP and non-TCP in ConnectivityMapQuery , all istio's policies conns (not only from sidecar) were considered TCP. (the splitting occurs only if there is an istio layer, so I thought that all captured conns in that case were considered TCP and what is not captured will be added to the non-TCP section).

I restricted only DNSEntry to TCP, since sidecar itself does not restrict conns 
but serviceEntry does, since it supports only TCP protocols ( when we add the support of ports field this will be more clear) 
so we don't want to see "DNS entries" in the non-TCP section

Reply:
As I understand from `def split_to_tcp_and_non_tcp_conns(conns):`  in `ConnectivityMapQuery`
it adds as a default `All Connections` for all protocols **other than TCP for** : for all the peers pairs which were captured in the `istio policies` (the captured ones are presented in the TCP section)
and by default all connections are added for both sections (TCP and non-TCP) ,for the non-captured peers at all. 

Note: the `DNSEntry peers` are exceptional, as we don't want to see them in the `non-TCP` section at all (these peers are relevant only for `istio` layer and conns to them may be enabled only with `TCP`, not enabled for other protocols at all ), then we have to make sure they always return as default only TCP connections (and not all conns for all protocols)